Events/MissionCompletedEvent.cs

```diff
@@ -54,6 +60,11 @@ public MissionCompletedEvent(DateTime timestamp, long? missionid, string name, s
             this.reward = reward;
             this.commodityrewards = commodityrewards;
             this.donation = donation;
+            if (commodityrewards.Count > 0)
+            {
+                this.rewardCommodity = commodityrewards[0].commodity;
```

Comment:
For completed missions without a commodity reward, have you tested to make sure this won't crash for null values?